woocommerce/woocommerce
# Decrease total sales per product when an order is reversed

### All Submissions:






### Changes proposed in this Pull Request:

* Add new functions to decrement `total_sales` in an order for canceled, refunded, and deleted orders.
* Adds post meta to persist that we've recorded cancelation (`_recorded_cancelations`).
* Updates the method that increments `total_sales` to clear `_recorded_cancelations`, so that toggling an order between statuses will properly update the total.



Closes #23796 .

### How to test the changes in this Pull Request:

1. Create a new product, and observe its current `total_sales` value.
2. Add that product to your cart, and checkout.
3. Verify that the `total_sales` value was increased by the quantity of that product in your cart.
4. Change the status of the order to `canceled`. Verify that the `total_sales` value was decreased by the quantity of that product in the order.
5. Change the status of the order to `refunded`. Verify that the `total_sales` value of the product remains unchanged.
6. Change the status of the order to `processing`. Verify that the `total_sales` value of the product is increased by the quantity of that product in the order.
7. Trash and then permanently delete the order. Verify that the `total_sales` value of the product is decreased by the quantity of that product in the order.

### Other information:




### Changelog entry

> Refunds, cancelations, and deletions of an order will now cause `total_sales` for a product to be decremented by the quantity of the product in the order.
